apps/server/src/modules/room/api/service/room-permission.service.ts

```diff
@@ -84,4 +84,16 @@ export class RoomPermissionService {
 		const isAllowed = canDeleteRoom || (isOwnSchool && canAdministrateSchoolRooms);
 		return isAllowed;
 	}
+
+	public async getReadableRoomIdsForUser(user: User): Promise<EntityId[]> {
+		const roomAuthorizables = await this.roomMembershipService.getRoomMembershipAuthorizablesByUserId(user.id);
```

Comment:
this method  is not needed here. 
The aboive call is already called in uc just before the method, so to remove duplication, simply put the filter code below in the uc